modules/contacts/server/controllers/contacts.server.controller.js

```diff
@@ -45,6 +48,29 @@ exports.add = function(req, res) {
         });
       }
 
+      // Check if contact already exists
+      Contact.findOne({
+        $or: [
+          { users: [ req.body.friendUserId, req.user._id ] },
+          { users: [ req.user._id, req.body.friendUserId ] }
+        ]
+      }).exec(function(err, existingContact) {
+        if (err) return done(err);
+
+        if (existingContact) {
+          // Contact already exists!
+          return res.status(400).json({
+            message: errorHandler.getErrorMessageByKey('invalid-id')
```

Comment:
[409 Conflict](https://httpstatuses.com/409) seems a good choice:
http://stackoverflow.com/a/3826024/2922612
http://stackoverflow.com/a/25541368/2922612

And I agree about the message, I'll fix something up.